### PR TITLE
refactor(date)!: unify signatures

### PR DESCRIPTION
**Pull request by Shinigami92**
related to https://github.com/faker-js/faker/issues/1349#issuecomment-1303820190

### CODE REVIEW

**Comment by Shinigami92** [comment]
Waiting on #1536 to be merged
- #1536

**Comment by Shinigami92** [comment]
Tests running green without touching them :heavy_check_mark: 

Will now add the new test cases :+1:

File: test/date.spec.ts [hunk 1]
```diff
@@ -19,14 +19,26 @@ describe('date', () => {
...
-      t.it('with only string refDate', undefined, refDate)
-        .it('with only Date refDate', undefined, new Date(refDate))
-        .it('with value', 10, refDate)
-        .it('with only number refDate', undefined, new Date(refDate).getTime());
+      t.it('with only string refDate', { refDate })
+        .it('with only Date refDate', { refDate: new Date(refDate) })
+        .it('with value', { years: 10, refDate })
+        .it('with only number refDate', {
+          refDate: new Date(refDate).getTime(),
+        });
```
Comment (ST-DDT) [inline test/date.spec.ts]: IMO these lines should be swapped as there are three `with only xyz refDate` and the other one is `with years`.

File: test/date.spec.ts [hunk 2]
```diff
@@ -19,14 +19,26 @@ describe('date', () => {
...
+    t.describeEach(
+      'recent',
+      'soon'
+    )((t) => {
+      t.it('with only string refDate', { refDate })
+        .it('with only Date refDate', { refDate: new Date(refDate) })
+        .it('with value', { days: 10, refDate })
+        .it('with only number refDate', {
+          refDate: new Date(refDate).getTime(),
+        });
```
Comment (ST-DDT) [inline test/date.spec.ts]: Same here (and below)

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
Does this need to be mentioned in the upgrading docs?

**Review by ejcheng — APPROVED** [review]

**Pull request merged**